pkg/storage/chunk/chunk.go

```diff
@@ -76,13 +86,16 @@ type Iterator interface {
 // 1 to 128.
 const BatchSize = 12
 
-// Batch is a sorted set of (timestamp, value) pairs.  They are intended to be
-// small, and passed by value.
+// Batches are sorted sets of (timestamp, value) pairs, where all values are of the same type (i.e. floats/histograms).
+//
+//	They are intended to be small, and passed by value!
 type Batch struct {
-	Timestamps [BatchSize]int64
-	Values     [BatchSize]float64
-	Index      int
-	Length     int
+	Timestamps    [BatchSize]int64
+	Values        [BatchSize]float64
+	PointerValues [BatchSize]unsafe.Pointer
```

Comment:
There was a version like that, but it leads to memory corruption since the code that uses the batches expects the batch to be copy-able. So we'd have to rewrite that part as well to make sure it copies the content instead which means another complication and loop....

Reply:
Thanks for the explanation. Resolving this comment.